nixui/options/api.py

```diff
@@ -182,3 +182,31 @@ def get_option_value(option):
         print('no default or value found for')
         print(option)
         print(get_leaf(option))
+
+
+###############
+# Apply Updates
+###############
+def apply_updates(option_value_obj_map):
+    """
+    option_value_obj_map: map between option string and python object form of value
+    """
+    option_expr_map = {
+        tuple(option.split('.')): object_to_expression.get_formatted_expression(value_obj)
```

Comment:
Currently, we don't support submodules, but this might not work for submodules or even `attrsOf`.

I have these lines on my configuration, and this can fail on these I think:
```nix
{
  boot = {
    kernel.sysctl."vm.overcommit_memory" = 1;
    kernel.sysctl."net.ipv4.ip_forward" = 1;
  };

  environment.etc = {
    "OpenCL/vendors/pocl.icd" = {
      source = ''${pkgs.pocl}/etc/OpenCL/vendors/pocl.icd'';
    };
  };
}
```

Reply:
Hmm, we have this code in our sample configuration:
```
  environment.etc = {
    "resolv.conf".text = ''
      nameserver 127.0.0.1  # unbound dns
      options edns0
    '';
  };
```

I'll write a test case for overwriting it, among other options.

Reply:
Yes, you're right. all but the commented test work in https://github.com/nix-gui/nix-gui/pull/47/commits/2bf53dd9b58fdf79c86e623c7f8810130074f116